src/Compilers/Core/Portable/InternalUtilities/NoThrowStream.cs

```diff
@@ -0,0 +1,53 @@
+﻿
+using System;
+using System.Diagnostics;
+using System.IO;
+
+namespace Roslyn.Utilities
+{
+    internal class NoThrowStreamDisposer : IDisposable
```

Comment:
Consider the name `DiagnosticStreamDisposer`.  Ignore this comment if you like 😄